middleware/csrf/csrf.go

```diff
@@ -98,12 +81,33 @@ func New(config ...Config) fiber.Handler {
 					HTTPOnly: cfg.CookieHTTPOnly,
 					SameSite: cfg.CookieSameSite,
 				})
-
 				return cfg.ErrorHandler(c, err)
 			}
 
 			// The token is validated, time to delete it
 			manager.delete(token)
+			generateToken = true
+		}
+
+		if generateToken {
```

Comment:
I think you change the original logic of this middleware. When using safe methods, and we already get token, we should get a new cookie with new expiration, And then we call `c.Vary(fiber.HeaderCookie)`.

Reply:
If you update the cookie here it will fail to validate later as you weren’t updating the manager storage ttl. The token is compared to stored value and not just the cookie in your code.

Reply:
ie you're extending the expiry of one element for the compare but not the other.

Reply:
It would be easy enough to extend the token in manager.setRaw(token, dummyValue, cfg.Expiration) by calling it on every request along with setting the cookie. That way both get expirations extended each request.

That would have performance implications, do we accept those?

Reply:
assume yes, commit made